solver/app/solproc.go

```diff
@@ -0,0 +1,86 @@
+package app
+
+import (
+	"context"
+
+	"github.com/omni-network/omni/anchor/anchorinbox"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/solutil"
+	"github.com/omni-network/omni/lib/umath"
+	"github.com/omni-network/omni/lib/xchain"
+	"github.com/omni-network/omni/solver/job"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+)
+
+func NewSolStreamCallback(
+	cl *rpc.Client,
+	chainVer xchain.ChainVersion,
+	cursors *cursors,
+	jobDB *job.DB,
+	asyncWork asyncWorkFunc,
+) solutil.StreamCallback {
+	return func(ctx context.Context, sig *rpc.TransactionSignature) error {
```

Comment:
solana streams all txs that touch an account. these can contain any instruction includig failed txs and including possible malicous events, so ignore those.